tests/model_selection/test_random_split.py

```diff
@@ -0,0 +1,178 @@
+#  Copyright 2023 MTS (Mobile Telesystems)
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import typing as tp
+from copy import deepcopy
+
+import numpy as np
+import pandas as pd
+import pytest
+
+from rectools import Columns
+from rectools.dataset import Interactions
+from rectools.model_selection import RandomSplitter
+
+
+class TestRandomSplitter:
+    @pytest.fixture
+    def shuffle_arr(self) -> np.ndarray:
+        return np.random.choice(np.arange(11), 11, replace=False)
+
+    @pytest.fixture
+    def interactions(self, shuffle_arr: np.ndarray) -> Interactions:
+        df = pd.DataFrame(
+            [
+                [1, 1, 1, "2021-09-01"],
+                [1, 2, 1, "2021-09-02"],
+                [2, 1, 1, "2021-09-02"],
+                [2, 2, 1, "2021-09-03"],
+                [3, 2, 1, "2021-09-03"],
+                [3, 3, 1, "2021-09-03"],
+                [3, 4, 1, "2021-09-04"],
+                [1, 2, 1, "2021-09-04"],
+                [3, 1, 1, "2021-09-05"],
+                [4, 2, 1, "2021-09-05"],
+                [3, 3, 1, "2021-09-06"],
+            ],
+            columns=[Columns.User, Columns.Item, Columns.Weight, Columns.Datetime],
+        ).astype({Columns.Datetime: "datetime64[ns]"})
+        return Interactions(df.iloc[shuffle_arr])
+
+    @pytest.fixture
+    def test_size(self) -> float:
+        return 0.25
+
+    @pytest.mark.parametrize("execution_number", range(5))
```

Comment:
how this `execution_number` parameter is used?

Reply:
It is used to repeat tests several times and compensate their randomness. As an alternative it is possible to set `random_state` in all tests to get reproducible results.

Reply:
It's not a good idea to run tests multiple time to get reproducible results because in some tiny number of cases we still can get different result

Reply:
`execution_number` removed